src/Entity/Course.php

```diff
@@ -15,181 +16,167 @@ class Course
+     * )
+     * @var ArrayCollection
      */
-    private $images = [];
+    private $images;
 
     /**
      * @ORM\Column(type="text")
+     * @var string
      */
     private $content;
 
     /**
-     * @ORM\Column(type="array", nullable=true)
+     * Owning side
+     *
+     * @ORM\ManyToMany(targetEntity="Rating")
+     * @ORM\JoinTable(name="courses_rating",
+     *     joinColumns={@ORM\JoinColumn(name="course_id", referencedColumnName="id", onDelete="CASCADE")},
+     *     inverseJoinColumns={@ORM\JoinColumn(name="rating_id", referencedColumnName="id", onDelete="CASCADE")}
+     * )
+     * @var ArrayCollection
      */
-    private $ratings = [];
+    private $ratings;
 
     /**
-     * @ORM\Column(type="array")
+     * Owning side
+     *
+     * @ORM\ManyToMany(targetEntity="Tag")
+     * @ORM\JoinTable(name="courses_tag",
+     *     joinColumns={@ORM\JoinColumn(name="course_id", referencedColumnName="id", onDelete="CASCADE")},
+     *     inverseJoinColumns={@ORM\JoinColumn(name="tag_id", referencedColumnName="id", onDelete="CASCADE")}
+     * )
+     * @var ArrayCollection
      */
-    private $tags = [];
+    private $tags ;
 
     /**
-     * @ORM\Column(type="object")
+     * Owning side
+     *
+     * @ORM\OneToOne(targetEntity="Summary")
+     * @var Summary
      */
     private $summary;
 
-    public function getId(): ?int
-    {
-        return $this->id;
+    public function __construct(
+        string $title,
+        string $description,
+        float $price,
+        float $duration,
+        string $content
+    ) {
+        $this->title = $title;
+        $this->description = $description;
+        $this->price = $price;
+        $this->addresses = new ArrayCollection();
+        $this->duration = $duration;
+        $this->images = new ArrayCollection();
+        $this->content = $content;
+        $this->ratings = new ArrayCollection();
+        $this->tags = new ArrayCollection();
+        $this->summary = new ArrayCollection();
     }
 
-    public function getTitle(): ?string
+    public function getId(): int
     {
-        return $this->title;
+        return $this->id;
     }
 
-    public function setTitle(string $title): self
+    public function getTitle(): string
     {
-        $this->title = $title;
-
-        return $this;
+        return $this->title;
     }
 
-    public function getDescription(): ?string
+    public function getDescription(): string
     {
         return $this->description;
     }
 
-    public function setDescription(string $description): self
-    {
-        $this->description = $description;
-
-        return $this;
-    }
-
-    public function getPrice(): ?float
+    public function getPrice(): float
     {
         return $this->price;
     }
 
-    public function setPrice(float $price): self
-    {
-        $this->price = $price;
-
-        return $this;
-    }
-
-    public function getAddresses(): ?array
+    public function getAddresses(): ArrayCollection
```

Comment:
toto by si moc nemal robiť, takto si vystavuješ celú kolekciu adries pre túto konkrétnu entitu a môžeš tak z vonku (napr.) odstrániť nejakú adresu a pri `flush()` sa ti zmena uloží do DB!

použi `return $this->addresses->toArray()`, návratový typ `: array` a dopíš si anotáciu...

to isté platí pre gettery nižšie